src/ikernel.ts

```diff
@@ -404,6 +417,17 @@ namespace IKernel {
      * The done handler for the kernel future.
      */
     onDone: () => void;
+
+    /**
+     * Register hook for IOPub messages.
+     * 
+     * @returns a disposable to unregister the hook.
+     * 
+     * #### Notes
+     * The most recently-registered hook is run first.
+     */
+    registerMessageHook(hook: (msg: KernelMessage.IIOPubMessage) => boolean): void;
```

Comment:
This could return an IDisposable, then registerMessageHook on the kernel would just return this value.